src/main/java/uk/gov/hmcts/reform/managecase/repository/IdamRepository.java

```diff
@@ -28,5 +31,10 @@ public UserInfo getUserInfo(String bearerToken) {
     public String getSystemUserAccessToken() {
         return idamClient.getAccessToken(appParams.getIdamSystemUserId(), appParams.getIdamSystemUserPassword());
     }
+
+    public List<UserDetails> searchUsers(String bearerToken, String query) {
+        return idamClient.searchUsers(bearerToken, query);
+    }
```

Comment:
This method should ideally take only user id and return the user object. Then, the method would handle all unexpected scenarios like multiple or no user objects being returned. Basically, anything that should go under data retrieval would be here. Current distribution of responsibilities among the repository and service layers is too light on the repo, and unfairly heavy on the service side.

Reply:
Addressed as mentioned in the above comment.